DataFormats/PortableTestObjects/src/classes.cc

```diff
@@ -0,0 +1,52 @@
+#define G__DICTIONARY
+
+#include "Rtypes.h"
+#include "RtypesImp.h"
+#include "TGenericClassInfo.h"
+#include "TVirtualObject.h"
+
+#include "classes.h"
+
+namespace ROOT {
+   // forward declaration
+   TGenericClassInfo *GenerateInitInstance(portabletest::TestHostCollection const*);
+}
+
+// read function for portabletest::TestHostCollection
+// called for every event
+static void setROOTReadStreamerFor_portabletest_TestHostCollection(char *target, TVirtualObject *oldObj)
+{
+  // address in memory of the buffer containing the object being read from file
+  char *onfile_address = (char *)oldObj->GetObject();
+  // offset of the "layout_" data member
+  static ptrdiff_t onfile_offset = oldObj->GetClass()->GetDataMemberOffset("layout_");
+  // reference to the Layout object being read from file
+  portabletest::TestHostCollection::Layout & onfile(*(portabletest::TestHostCollection::Layout *)(onfile_address + onfile_offset));
+  // pointer to the portabletest::TestHostCollection object being constructed in memory
+  portabletest::TestHostCollection *newObj = (portabletest::TestHostCollection *)target;
+
+  portabletest::TestHostCollection::ROOTReadStreamer(newObj, onfile);
+}
+
+// set the read function for portabletest::TestHostCollection
+// called when the dictionary is loaded
+static bool set_portabletest_TestHostCollection_ReadRules()
+{
+   // build the read rules
+   std::vector<::ROOT::Internal::TSchemaHelper> readrules(1);
+   ::ROOT::Internal::TSchemaHelper *rule = &readrules[0];
+   rule->fSourceClass = "portabletest::TestHostCollection";
+   rule->fSource = "portabletest::TestHostCollection::Layout layout_;";
+   rule->fTarget = "";
+   rule->fFunctionPtr = (void *)TFunc2void(setROOTReadStreamerFor_portabletest_TestHostCollection);
+   rule->fCode = "";
```

Comment:
This is could be set to `"portabletest::TestHostCollection::ROOTReadStreamer(newObj, onfile.layout_);"`.
For the moment it is left empty on purpose, in case we decide to modify `setROOTReadStreamerFor_portabletest_TestHostCollection()` and `portabletest::TestHostCollection::ROOTReadStreamer()` in a way that cannot be reflected in the XML read rule.